lib/superagent-mock.js

```diff
@@ -142,7 +142,7 @@ function mock (superagent, config, logger) {
       }
     }
 
-    fn(error, response);
```

Comment:
Even if it's longer, I would prefer to not run `setTimeout` if there is no delay 😉 
```js
if (context.delay) {
    setTimeout(function() {
        fn(error, response);
    }, context.delay);
} else {
    fn(error, response);
}
```